### PR TITLE
Phy test tool sample update

### PR DESCRIPTION
**Pull request by elwi16**
Due to the automatisation of phy test tool tests, two more configurations are added to the sample.yaml file

### CODE REVIEW

**Comment by jciupis** [comment]
Please reword the commit messages to follow contribution guidelines: https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines

File: samples/peripheral/802154_phy_test/sample.yaml [hunk 1]
```diff
@@ -9,4 +9,22 @@ tests:
       - nrf21540dk_nrf52840
       - nrf5340dk_nrf5340_cpunet
     platform_allow: nrf52840dk_nrf52840 nrf21540dk_nrf52840 nrf5340dk_nrf5340_cpunet
-    tags: ci_build
+    tags: ci_rs_build
```
Comment (dawidprzybylo) [inline samples/peripheral/802154_phy_test/sample.yaml:12]: tag should be ci_build as original sample, you can add `ci_rs_build` as second tag for our use

File: samples/peripheral/802154_phy_test/sample.yaml [hunk 2]
```diff
@@ -9,4 +9,22 @@ tests:
...
+  sample.peripheral.802154_phy_test_auto:
```
Comment (dawidprzybylo) [inline samples/peripheral/802154_phy_test/sample.yaml:13]: add antenna diversity in sample name, maybe `sample.peripheral.802154_phy_test.ant_div_mode_auto`

File: samples/peripheral/802154_phy_test/sample.yaml [hunk 3]
```diff
@@ -9,4 +9,22 @@ tests:
...
+    build_only: true
+    integration_platforms:
+      - nrf21540dk_nrf52840
+      - nrf52840dk_nrf52840
+    platform_allow: nrf21540dk_nrf52840 nrf52840dk_nrf52840
+    tags: ci_rs_build
+    extra_args: DCONFIG_PTT_ANTENNA_DIVERSITY=y
+      DCONFIG_PTT_ANT_MODE_AUTO=y
+  sample.peripheral.802154_phy_test_manual:
```
Comment (dawidprzybylo) [inline samples/peripheral/802154_phy_test/sample.yaml:22]: name should be changed with antenna diversity feature include: `sample.peripheral.802154_phy_test.ant_div_mode_manual`

File: samples/peripheral/802154_phy_test/sample.yaml (hunk 2 above)
Comment (dawidprzybylo) [reply samples/peripheral/802154_phy_test/sample.yaml:13]: Moreover Antenna diversity is valid only for nrf21540dk platform that has phisicall two antennas
nrf52840dk is incorrect in this situation (valid only if FEM EK is added)

File: samples/peripheral/802154_phy_test/sample.yaml [hunk 4]
```diff
@@ -9,4 +9,22 @@ tests:
       - nrf21540dk_nrf52840
       - nrf5340dk_nrf5340_cpunet
```
Comment (dawidprzybylo) [inline samples/peripheral/802154_phy_test/sample.yaml:10]: Should we add nrf5340dk_nrf5340_cpuapp variant here? @ankuns @ahasztag

File: samples/peripheral/802154_phy_test/sample.yaml (hunk 3 above)
Comment (dawidprzybylo) [reply samples/peripheral/802154_phy_test/sample.yaml:22]: Sample valid only for nrf21540dk platform

**Review by dawidprzybylo — CHANGES_REQUESTED** [review]

File: samples/peripheral/802154_phy_test/sample.yaml (hunk 1 above)
Comment (dawidprzybylo) [reply samples/peripheral/802154_phy_test/sample.yaml:12]: If you want to add second tag, please add it accordingly base which tag is used in RS cycle, and when we want to build it

File: samples/peripheral/802154_phy_test/sample.yaml (hunk 3 above)
Comment (elwi16) [reply samples/peripheral/802154_phy_test/sample.yaml]: fixed

File: samples/peripheral/802154_phy_test/sample.yaml (hunk 2 above)
Comment (elwi16) [reply samples/peripheral/802154_phy_test/sample.yaml]: fixed :)

File: samples/peripheral/802154_phy_test/sample.yaml (hunk 1 above)
Comment (elwi16) [reply samples/peripheral/802154_phy_test/sample.yaml]: added new tags for integration and weekly

File: samples/peripheral/802154_phy_test/sample.yaml [hunk 5]
```diff
@@ -9,4 +9,20 @@ tests:
...
     platform_allow: nrf52840dk_nrf52840 nrf21540dk_nrf52840 nrf5340dk_nrf5340_cpunet
-    tags: ci_build
+    tags: ci_build ci_rs_weekly
+  sample.peripheral.802154_phy_test.ant_div_mode_auto:
+    build_only: true
+    integration_platforms:
+      - nrf21540dk_nrf52840
+    platform_allow: nrf21540dk_nrf52840
+    tags: ci_build ci_rs_weekly
+    extra_args: DCONFIG_PTT_ANTENNA_DIVERSITY=y
```
Comment (dawidprzybylo) [inline samples/peripheral/802154_phy_test/sample.yaml:19]: Remove `D` prefix from any Kconfig here, twister will add it underneath. please also manually verify via twister if all samples could be builded using zephyr twister tool

**Review by dawidprzybylo — CHANGES_REQUESTED** [review]

File: samples/peripheral/802154_phy_test/sample.yaml (hunk 4 above)
Comment (jciupis) [reply samples/peripheral/802154_phy_test/sample.yaml:10]: No, this sample will not build for `nrf5340dk_nrf5340_cpuapp`. Only the current variant should stay

**Comment by jciupis** [comment]
I suggest changing the commit message as it currently states that new samples were implemented, which is misleading.

File: samples/peripheral/802154_phy_test/sample.yaml (hunk 5 above)
Comment (elwi16) [reply samples/peripheral/802154_phy_test/sample.yaml]: I removed the prefixes, manual build works :)

File: samples/peripheral/802154_phy_test/sample.yaml [hunk 6]
```diff
@@ -9,4 +9,20 @@ tests:
       - nrf21540dk_nrf52840
       - nrf5340dk_nrf5340_cpunet
     platform_allow: nrf52840dk_nrf52840 nrf21540dk_nrf52840 nrf5340dk_nrf5340_cpunet
-    tags: ci_build
+    tags: ci_rs_build ci_rs_weekly
```
Comment (jciupis) [inline samples/peripheral/802154_phy_test/sample.yaml:12]: Have you removed the `ci_build` tag intentionally?

Comment (dawidprzybylo) [reply samples/peripheral/802154_phy_test/sample.yaml:12]: ci_build please change it to `ci_build ci_rs_weekly`

File: samples/peripheral/802154_phy_test/sample.yaml [hunk 7]
```diff
@@ -9,4 +9,20 @@ tests:
...
+  sample.peripheral.802154_phy_test.ant_div_mode_auto:
+    build_only: true
+    integration_platforms:
+      - nrf21540dk_nrf52840
+    platform_allow: nrf21540dk_nrf52840
+    tags: ci_rs_build ci_rs_weekly
```
Comment (dawidprzybylo) [inline samples/peripheral/802154_phy_test/sample.yaml:18]: please add original twister tag `ci_build` here

File: samples/peripheral/802154_phy_test/sample.yaml [hunk 8]
```diff
@@ -9,4 +9,20 @@ tests:
...
+    extra_args: CONFIG_PTT_ANTENNA_DIVERSITY=y
+      CONFIG_PTT_ANT_MODE_AUTO=y
+  sample.peripheral.802154_phy_test.ant_div_mode_manual:
+    build_only: true
+    integration_platforms:
+      - nrf21540dk_nrf52840
+    platform_allow: nrf21540dk_nrf52840
+    tags: ci_rs_build ci_rs_weekly ci_rs_integration
```
Comment (dawidprzybylo) [inline samples/peripheral/802154_phy_test/sample.yaml:26]: please add original twister tag `ci_build` here

**Review by dawidprzybylo — CHANGES_REQUESTED** [review]

File: samples/peripheral/802154_phy_test/sample.yaml (hunk 6 above)
Comment (elwi16) [reply samples/peripheral/802154_phy_test/sample.yaml]: I updated now, thanks!

**Review by jciupis — APPROVED** [review]

**Review by ahasztag — APPROVED** [review]

**Review by dawidprzybylo — APPROVED** [review]

**Review by ankuns — APPROVED** [review]

**Pull request merged**